glib/src/thread_pool.rs

```diff
@@ -77,11 +78,12 @@ impl ThreadPool {
     pub fn push_future<T: Send + 'static, F: FnOnce() -> T + Send + 'static>(
```

Comment:
Hm, we could have `push()` above also return a `JoinHandle` kind of thing around a `std::sync::mpsc` channel, I guess

Reply:
I think that would make it identical to `push_future`

Reply:
It would return a blocking `JoinHandle` like `std::thread::spawn`

Reply:
Done, feeling a bit better about using that after seeing this land for 1.67: https://github.com/rust-lang/rust/pull/93563